python_modules/libraries/dagster-dg/dagster_dg/mcp/server.py

```diff
@@ -5,126 +5,242 @@
 
 
 @mcp.tool()
-async def scaffold_dagster_project(project_path: str, project_name: str) -> str:
-    """Create a new Dagster project at the location of `project_path` and the name `project_name`.
+async def scaffold_dagster_project(project_path: str) -> str:
+    """Create a new Dagster project with the name `project_name` in the cwd.
+
+    Conventionally, the folder name is the project name.
 
     Args:
-        project_path: The path to your Dagster project.
+        project_path: The absolute path to the project to be scaffolded.
 
     Returns:
         The output of the project scaffold command.
     """
-    return subprocess.check_output(
-        ["uv", "run", "dg", "scaffold", "project", project_path], stderr=subprocess.STDOUT
-    ).decode("utf-8")
+    try:
+        return subprocess.check_output(
+            ["uv", "run", "dg", "scaffold", "project", project_path],
+            stderr=subprocess.STDOUT,
+        ).decode("utf-8")
+    except subprocess.CalledProcessError as e:
+        raise Exception("Subprocess failed", e.output)
 
 
 @mcp.tool()
-async def list_dagster_components(project_path: str) -> str:
+async def list_available_components(project_path: str) -> str:
     """List all Dagster components for the project.
-    Components can be scaffolded to create new Dagster definitions. Call this
-    when you need to add new Dagster definitions and want to get a list
-    of components to use to do so.
+
+    Components can be scaffolded to create new Dagster definitions. Call this when you need to add
+    new Dagster definitions and want to get a list of components to use to do so.
 
     Args:
-        project_path: The path to your Dagster project.
+        project_path: The full path to your Dagster project.
 
     Returns:
         The list of components available for the given Dagster project.
     """
-    return subprocess.check_output(
-        [
-            "uv",
-            "run",
-            "dg",
-            "list",
-            "plugins",
-            "--feature",
-            "component",
-        ],
-        cwd=project_path,
-        stderr=subprocess.STDOUT,
-    ).decode("utf-8")
+    try:
+        return subprocess.check_output(
+            [
+                "uv",
+                "run",
+                "dg",
+                "list",
+                "plugins",
+                "--feature",
+                "component",
+                "--json",
+            ],
+            cwd=project_path,
+            stderr=subprocess.STDOUT,
+        ).decode("utf-8")
+    except subprocess.CalledProcessError as e:
+        raise Exception("Subprocess failed", e.output)
```

Comment:
@benpankow we were suppressing the error message, so temporarily I'm exposing `e.output` for the agent to interpret.